doc/Kernel_Attributes.md

```diff
@@ -0,0 +1,122 @@
+Kernel Attributes
+=================
+
+Kernel attributes are stored in a data structure at the end of the kernel's
+allocated flash region. These attributes describe properties of the flashed
+kernel on a particular hardware board. External tools can read these attributes
+to learn about the kernel installed on the board.
+
+<!-- npm i -g markdown-toc; markdown-toc -i Kernel_Attributes.md -->
+
+<!-- toc -->
+
+- [Format](#format)
+  * [Header Format](#header-format)
+  * [TLV Format](#tlv-format)
+- [TLVs](#tlvs)
+  * [App Memory (0x0101)](#app-memory-0x0101)
+  * [Kernel Binary (0x0102)](#kernel-binary-0x0102)
+- [Kernel Attributes Location](#kernel-attributes-location)
+
+<!-- tocstop -->
+
+## Format
+
+Kernel attributes are stored in a descending TLV (type-length-value) structure.
+That means they start at the highest address in flash, and are appended in
+descending flash addresses.
```

Comment:
Last minute question: what about alignment? I don't think TLVs have any size-alignment constraints, and byte-aligned reads might make reads from within Tock more expensive?

Reply:
Why would tock read this? It already has the linker values in the code. This is for tockloader.